napari/_qt/qt_main_window.py

```diff
@@ -81,9 +84,12 @@ class _QtMainWindow(QMainWindow):
     # *no* active windows, so we want to track the most recently active windows
     _instances: ClassVar[List['_QtMainWindow']] = []
 
-    def __init__(self, viewer: 'Viewer', parent=None) -> None:
+    def __init__(
+        self, viewer: 'Viewer', window: 'Window', parent=None
+    ) -> None:
```

Comment:
Why not get `Window` from `Viewer`? This pattern suggests that we could pass `window` with unrelated `viewer`

Reply:
yeah, I'll change this. 

> This pattern suggests that we could pass window with unrelated viewer

that actually will make sense in some/many cases.  For example, a "show full screen" or "close window" action really doesn't care about a viewer

Reply:
Just wanted to add reference here: https://github.com/napari/napari/pull/4865/files#r986462579